### PR TITLE
feat: parallelizing sending bulk updates to OpenSearch on same snapshot and updating tessellation version

### PR DESCRIPTION
**Pull request by marcus-girolneto**
### Changes
+ To address recent issues with snapshot streaming when sending data to OpenSearch, the bulk update process has been optimized by introducing parallelization.
+ Now, updates for each snapshot are sent in parallel while ensuring that only one snapshot is processed at a time.
+ This optimization helped improve the performance on 2024-12-02, helping the system catch up with the network.

### Tests

Screenshot 2024-12-03 at 10 38 35

### NOTE
**I've generated a JAR with these changes from version 4.3.3 and deployed it to the SnapshotStreaming of Mainnet, @AlexBrandes authorized this since we needed to do something to catch up with the network, so we need to generate an official version and maybe re-deploy**

### CODE REVIEW

File: src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala [hunk 1]
```diff
@@ -119,7 +119,18 @@ object SnapshotProcessor {
...
+            for {
+              start <- Clock[F].realTime
+              _ <- logger.info(s"Starting to send bulk updates to Opensearch")
+              result <- requests.parTraverse { br =>
+                opensearchDAO.sendToOpensearch(bulk(br))
+              }
+              end <- Clock[F].realTime
+              elapsedTime = end.toMillis - start.toMillis
+              _ <- logger.info(s"Bulk update operation took $elapsedTime ms")
+            } yield result
```
Comment (scasplte2) [inline src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala]: I believe cats provides a method for timing that does this for you

``` scala
logger.info(s"Starting to send bulk updates to Opensearch") >>
requests.parTraverse { br =>
  opensearchDAO.sendToOpensearch(bulk(br))
}.timed.flatTap { case (elapsedTime, _) =>
  logger.info(s"Bulk update operation took ${elapsedTime.toMillis} ms")
}
```

File: project/Dependencies.scala [hunk 2]
```diff
@@ -76,7 +76,7 @@ object Dependencies {
 
     val organizeImports = "com.github.liancheng" %% "organize-imports" % V.organizeImports
 
-    val tessellationSdk = "org.constellation" %% "tessellation-sdk" % V.tessellation
+    val tessellationNodeShared = "org.constellation" %% "tessellation-node-shared" % V.tessellation
```
Comment (scasplte2) [inline project/Dependencies.scala]: Why the switch? I thought Marcin was urging us to move to only using `sdk` and this change doesn't use any additional tooling from Tessellation.

File: src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala (hunk 1 above)
Comment (marcus-girolneto) [reply src/main/scala/org/constellation/snapshotstreaming/SnapshotProcessor.scala]: 👍🏼

File: project/Dependencies.scala (hunk 2 above)
Comment (marcus-girolneto) [reply project/Dependencies.scala]: 👍🏼

**Review by scasplte2 — APPROVED** [review]
LGTM

**Pull request merged**